setup.py

```diff
@@ -37,7 +37,8 @@ def find_version(*file_paths):
 
 install_deps = [
     'pillow>=4.0.0',
-    'smbus2'
+    'smbus2',
+    'pyftdi;python_version>="3.5"'
```

Comment:
this is an optional dependency.. looking at pyftdi, it depends on

```
INSTALL_REQUIRES = [
    'pyusb >= 1.0.0',
    'pyserial >= 3.0',
]
```

so installing luma.core would also pull in these dependencies. I'd rather not do that.. thoughts?

Reply:
Surely pyftdi is optional in the same way smbus or spidev are both optional - if you only want to run an I2C display, it still installs spidev for example.

Reply:
> if you only want to run an I2C display, it still installs spidev for example.

if you only want to run an i2c display, it will install pyusb and pyserial that you do not need. every luma library (depends on core) will now install pyserial and pyusb that you do not need..

Reply:
So what is the alternative? Not pyftdi a dependency and at runtime cause luma.core to crash with an import error? That makes for a crappy user experience and an extra burden on fielding github issues, for the only benefit of a few less files that were installed as a transitive dependency.

One could argue that there are many libraries installed (both at the OS level, and within the python installation directories) that you do not need, but they are installed anyway. 

We aren’t talking about devices that are particularly memory- or storage-constrained, and if you are concerned about polluting your main python dist-packages or site-packages with unnecessary cruft, then surely there’s virtualenv/pipenv/etc to manage all that.

I think I’m just unsure what the hesitation is about pulling in those dependencies verses the trade-offs of not doing so, ... and then the complexity of having to manage installing them somewhere else.

Reply:
> Not making pyftdi a dependency and at runtime cause luma.core to crash with an import error?

I wouldn't call it a crash tbh. It's like the luma.examples that look for psutil and warn if it's not installed. I prefer to keep things simple and lightweight and installing a dependency that is only used for a small part of luma.core feels like overkill. Adding these dependencies now (without a small 'fight') means we will keep bloating the library. This issue reminds me of the pygame dependency back in the day. We eventually separated it into a different library (luma.emulator), making it trivial to install luma.core/luma.oled because installing pygame (without an egg) is not easy and we didn't want to put this uncessary burden on the user if they do not even use it. I have a similar feeling with this. These dependencies could result in install hell. And what are we going to tell the user then, something like:

- git clone the repo
- manually remove the pyftdi dep from setup.py
- and `pip install -e .`?

maybe I am exagerating but less is more IMO.

What happens if you do a pyftdi installation in a clean RPI system? If it doesn't cause any issues, I am +1 for including this dep. If it causes issues we have to take a look at those because adding extra complexity to luma.core (which will affect all other luma libs) sounds like a really bad idea to me.

Reply:
> installing pygame (without an egg) is not easy

Good point .. had forgotten about that on RPi

> What happens if you do a pyftdi installation in a clean RPI system?

Will try tomorrow

Reply:
>  if you are concerned about polluting your main python dist-packages or site-packages with unnecessary cruft

it's more that you first need to `apt-get` a bunch of stuff before you can properly pip install some packages.

> Will try tomorrow

Thanks.